gateway-sp-comms/src/single_sp.rs

```diff
@@ -423,12 +417,10 @@ impl SingleSp {
         } else {
             MgsRequest::ComponentSetActiveSlot { component, slot }
         };
-        self.rpc(msg).await.and_then(|(_peer, response, _data)| {
-            if persist {
-                expect_component_set_and_persist_active_slot_ack(response)
-            } else {
-                expect_component_set_active_slot_ack(response)
-            }
+        self.rpc(msg).await.and_then(if persist {
+            expect_component_set_and_persist_active_slot_ack
+        } else {
+            expect_component_set_active_slot_ack
```

Comment:
Huh, I don't think I'd have thought to use an `if` evaluating to a `fn` like this. Neat!